### DISCUSSION TITLE
Question on the new Settings ( Text Encoder )

### DISCUSSION THREAD

**Discussion by metzo007** [TheLastBen/fast-stable-diffusion#971] [General]
I always skipped the Text_Encoder_Training since I never got any good results using it. Either the CKPT always produced similar images or not training related at all. 

Since the option to skip the Text_Encoder_Training training is gone, I wounder how to use the training without it now. Do I use all the steps with UNet_Training_Steps and leave the Text_Encoder_Training at zero, or was the old Text_Encoder_Training only the option to limit the Text_Encoder and I have to leave the UNet_Training_Steps at zero.
Results are not good, resulds are not following the prompt

Also, I found little Info on the purpose of the Text_Encoder_Training. What "Text" is it referring to? Why should I use it, and what it the difference?

I found a post on reddit saying, in order to make the Text_Encoder_Training work, I need to name the images like the prompt like "drawing of a house in the woods.jpg", but I doubt that.

**Comment by TheLastBen** [comment]
You should never caption the images when using the text_encoder, stick to the rule, use only one nonexistent word as the instance images filename.
set the text_encoder to 350-500 steps and the UNet steps to 3000, if the results aren't enough, resume training the UNet with 1000 steps and setting the text_encoder to 0 (disabled) 

if you disable the text_encoder the whole training, you will never get good results.

**Reply by metzo007** [reply]
Thanks for the quick response. In the only version I alway left the "Enable_text_encoder_training:" unchecked and good good results. I thought setting the text_encoder to zero would have the same effrect. 

I did a test run with running text_encoder only and the resulds are also not very good.

I would like to have settings similar to "Enable_text_encoder_training:" unchecked like in the old version. Also, is there no explanation on what the text_encoder does? 

Sorry for all the stupid questions ;-)

**Reply by TheLastBen** [reply]
when you set the text_encoder to 0 means it's disabled, you're getting good results without it because you caption the images, but that reduces diversity and overfits the model. the text encoder helps connect the text to image, when you refine it just the right amount, you will get great results.
